esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -866,19 +869,25 @@ void EsdlServiceImpl::generateTargetURL(IEspContext & context,
 
     StringBuffer name(srvinfo->queryProp("@name"));
 
-    ISmartSocketFactory *conn = connMap.getValue(name);
-    if (!conn)
+    CSmartSocketFactory *sconn = static_cast<CSmartSocketFactory*>(connMap.getValue(name));
+    if (!sconn)
         throw MakeStringException(-1, "Could not create smartsocket.");
 
     url.set(srvinfo->queryProp("@prot"));
     if (url.length() <= 0)
         url.append("HTTP");
     url.append("://");
 
-    SocketEndpoint ep;
-    ep = conn->nextEndpoint();
-
-    ep.getUrlStr(url);
+    SmartSocketEndpoint * sep = sconn->nextSmartEndpoint();
```

Comment:
Correct, this is the reason I instantiated the SmartSocket, I thought about adding it to the ISmartSocketFactory but it seemed like an implementation specific feature. The reason I wanted the SEP was to access the hostname. I'll add a method to query the next available hostname if available.